helix-term/src/commands/lsp.rs

```diff
@@ -252,6 +252,9 @@ fn diag_picker(
                 .into()
             },
         ),
+        ui::PickerColumn::new("source", |item: &PickerDiagnostic, _| {
+            item.diag.source.clone().unwrap_or("".to_string()).into()
```

Comment:
This string allocation can be avoided:

```suggestion
            item.diag.source.as_deref().unwrap_or("").into()
```

Reply:
Thanks, that is clearly better! I am a bit of a Rust novice :sweat_smile: